### PR TITLE
Heroku

### PR DESCRIPTION
**Pull request by ghadeerda**

### CODE REVIEW

**Review by alekla — CHANGES_REQUESTED** [review]
You should add *.pyc files into .gitignore

File: app.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+import logging
+from datetime import date
+
+server="127.0.0.1"
+#server="https://ur-opinion.herokuapp.com/"
+def test_local_server():
+    #if ( "127.0.0.1" in request.remote_addr ) :
+    if ("ur-opinion.herokuapp.com" in request.referrer ) :    
+        return True
+
```
Comment (alekla) [inline app.py]: Empty line, you should remove it.

File: app.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
+from flask import Flask, render_template,request,redirect
+import re
+import requests
+from logging.handlers import RotatingFileHandler
+import logging
+from datetime import date
+
+server="127.0.0.1"
+#server="https://ur-opinion.herokuapp.com/"
+def test_local_server():
```
Comment (alekla) [inline app.py:12]: Can you please explain, what we need this function?

File: app.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+from routes import *
+
+app = Flask(__name__)
+app.url_map.strict_slashes = False
+
+#secret key for the session
+app.secret_key = "super secret key"
+
+@app.route('/favicon.ico')
+def favicon():
```
Comment (alekla) [inline app.py:27]: You can move it to static routes

File: app.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    rp = request.path 
+    if rp != '/' and rp.endswith('/'):
+        return redirect(rp[:-1])
+    if rp.endswith("index"):
+        return redirect(rp[:-5])
+
+#main page
+@app.route('/')
+def index():
+
```
Comment (alekla) [inline app.py]: So too may empty lines, we need to add some linter to fix them.

File: app.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+          teamid = request.form.get('teamid').strip()
+          
+          removeteam_r(teamid)
+          session["msg"]="loadteams"
+          return redirect("/")
+       
+    else:
+        return redirect("/login", code=302)        
+    
+#when pressing new task button (to get the teams)
```
Comment (alekla) [inline app.py]: So empty comments better to remove.

File: app.py [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+          desc = request.form.get('taskdesc').strip()
+          teamid = request.form.get('slc_teams').strip()
+          # datepub = request.form.get('datepublish').strip()
+          #get current date from server
+          datepub = str(date.today())
+          eachperiod = request.form.get('eachperiod').strip()
+          
+          addtask_r(name , desc , teamid , datepub, eachperiod)
+          
+          session["msg"]="loadcreatedtasks"
```
Comment (alekla) [inline app.py]: Can you please explain why we need store some info into session?

File: app.py [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+def in_progress():
+           # use the host of the server
+          if (test_local_server()) :
+              return  in_progress_r() 
+              
+          return render_template('error.html'), 404
+          
+  
+#created by me  cards
+# show tasks that I have created with status 0 or 1 (finished or not)
```
Comment (alekla) [inline app.py]: It's very good, that you add some documentation,

File: app.py (hunk 2 above)
Comment (ghadeerda) [reply app.py:12]: We need it temporarily to prevent others access to sub_routes which are called to load into internal div.

**Review by alekla — CHANGES_REQUESTED** [review]

File: app.py (hunk 4 above)
Comment (alekla) [reply app.py]: So I will be add linter.

**Review by alekla — APPROVED** [review]

**Pull request closed without merging**